### ISSUE TITLE
readability-convert-member-functions-to-static: inspection doesn't handle case when template is used

### ISSUE THREAD

**Issue by maxim-banaev** [llvm/llvm-project#43654]
|  |  |
| --- | --- |
| Bugzilla Link | [44309](https://llvm.org/bz44309) |
| Version | unspecified |
| OS | All |
| CC | @EugeneZelenko,@mgehre |

## Extended Description

The inspection should be triggered for `getT()`:

```c++
template <typename T> class X {
  static T t;

public:
  int getT() { return t; }
};
```

**Comment by localspook** [comment]
The check doesn't warn here deliberately:
https://github.com/llvm/llvm-project/blob/b30034da0ffaf67a144d062607e8f627e14227d1/clang-tools-extra/test/clang-tidy/checkers/readability/convert-member-functions-to-static.cpp#L131-L137
